pkg/apis/machine/validation/alicloudmachineclass.go

```diff
@@ -63,6 +64,34 @@ func validateAlicloudMachineClassSpec(spec *machine.AlicloudMachineClassSpec, fl
 		allErrs = append(allErrs, field.Required(fldPath.Child("keyPairName"), "KeyPairName is required"))
 	}
 
+	if spec.DataDisks != nil {
+		names := map[string]int{}
+		for i, dataDisk := range spec.DataDisks {
+			idxPath := fldPath.Child("dataDisks").Index(i)
+			if _, keyExist := names[dataDisk.Name]; keyExist {
+				names[dataDisk.Name]++
+			} else {
+				names[dataDisk.Name] = 1
+			}
+
+			if dataDisk.Name == "" {
+				allErrs = append(allErrs, field.Required(idxPath.Child("name"), "Data Disk name is required"))
+			}
+			if dataDisk.Size <= 0 {
```

Comment:
For AliCloud, data disk size should be larger than 20 and less than 32768 GB.

Reply:
done